api/src/pcapi/core/finance/api.py

```diff
@@ -1667,6 +1703,11 @@ def _prepare_invoice_context(invoice: models.Invoice, batch: models.CashflowBatc
         period_start=period_start,
         period_end=period_end,
         reimbursements_by_venue=reimbursements_by_venue,
+        including_finance_incident=any(
+            cashflow
+            for cashflow in cashflows
+            if any(pricing.event.bookingFinanceIncident for pricing in cashflow.pricings)
```

Comment:
[détail / optimisation] Puisqu'ici on ne veut que vérifier la présence (sans avoir besoin de l'objet `BookingFinanceIncident`), on peut se contenter de vérifier `bookingFinanceIncidentId`, pour éviter un accès à l'objet (voire une requête SQL si on a oublié de "joinedload") :

```suggestion
            if any(pricing.event.bookingFinanceIncidentId for pricing in cashflow.pricings)
```